bonus.js

```diff
@@ -0,0 +1,67 @@
+/*
+Bonusopdracht
+    [v] b1.Maak drie knoppen op de pagina:
+    [v] Sorteer op prijs,
+    [v] AmbiLight TV's en
+    [v] Uitverkochte exemplaren.
+    [v] Gebruik de code die je in opdracht
+    [v] 1b hebt gemaakt en schrijf dit om naar functies
+    [v] 1c hebt gemaakt en schrijf dit om naar functies
+    [v] 1d hebt gemaakt en schrijf dit om naar functies
+    [v] zodat je ze kunt aanroepen op het moment dat de buttons geklikt worden.
+    [v] Zorg ervoor dat de functies de uitkomsten in de de console loggen als de gebruiker op de bijbehorende knop klikt.
+    [v] Tip: lees hiervoor paragraaf 7.4 op EdHub eens door!
+    [v] b2.Zorg er nu voor, in plaats van dat de uitkomsten in de console worden gelogd, dat de uitkomsten worden meegegeven
+        aan jouw generateTV-functie zodat de resultaten daadwerkelijk op de pagina weergegeven worden!
+*/
+console.log("bonus");
+
+//bonus1.
+
+console.log("uit komst op bonus1: ");
+
+//bonus1b.
+function productUitverkocht(object) {
+
+    const uitverkocht = [];
+
+    object.map((inventory) => {
+        if ((inventory.originalStock - inventory.sold) === 0) {
+
+            uitverkocht.push(inventory);
+        }
+    });
```

Comment:
Een map-functie heeft een return waarde. Zoals hij hier gebruikt wordt lijkt het op een forEach-functie. Je zou zoiets kunnen doen:
```suggestion
    const uitverkocht = object.map(inventory => inventory.originalStock === inventory.sold);
```